frontend/src/component/PublicProfile/CustomFormManager.jsx

```diff
@@ -0,0 +1,311 @@
+          styleLabel={{ top: '8px', cursor: 'default' }}
+          styleCheck={{ top: '-5px' }}
+        />
+      </span>
+    </div>
+  )
+}
+
+const DisplaySchemaArray = props => {
+  if (!props.valueList) return null
+
+  if (props.valueList.every(value => typeof value === 'string')) {
+    return (
+      <DisplaySchemaPropertyString
+        label={props.label}
+        value={props.valueList.join(', ')}
+      />
+    )
+  }
+
+  return [
+    <DisplayTitle
+      label={props.label}
+      key={props.label}
+    />,
+    props.valueList.map((value, i) => {
+      if (!value) return null
+
+      const valueType = typeof value
+
+      if (valueType === 'object') {
+        return (
+          <div
+            className='DisplaySchemaArray'
+            key={`object_${props.currentKey}_${value.title}_${i}`}
+          >
+            <DisplaySchemaObject
+              schemaObject={props.schemaObject}
+              dataSchemaObject={value}
+              key={`object_${props.currentKey}_${value.title}__DisplaySchemaObject_${i}`}
+            />
+          </div>
+        )
+      }
+
+      return null
+    })
+  ]
+}
+
+const DisplaySchemaObject = props => {
+  if (!props.dataSchemaObject || !props.schemaObject) {
+    console.log('Error in DisplaySchemaObject, null props', props)
+    return null
+  }
+
+  const title = props.schemaObject.title
+    ? (
+      <DisplayTitle
+        label={props.schemaObject.title}
+        key={props.schemaObject.title}
+      />
+    )
+    : null
+
+  return [
+    title,
+    <div className='DisplaySchemaObject' key={`object_root_${props.nestedLevel}`}>
+      {Object.entries(props.dataSchemaObject).map(([key, value]) => {
+        const valueType = typeof value
+
+        if (Array.isArray(value)) {
+          return (
+            <DisplaySchemaArray
+              label={props.schemaObject.properties[key].title}
+              valueList={value}
+              currentKey={key}
+              schemaObject={props.schemaObject.properties[key].items}
+              key={`array_${key}`}
+            />
+          )
+        }
+
+        if (valueType === 'object') {
+          return (
+            <DisplaySchemaObject
+              schemaObject={props.schemaObject.properties[key]}
+              dataSchemaObject={value}
+              nestedLevel={props.nestedLevel + 1}
+              key={`object_${key}`}
+            />
+          )
+        }
+
+        if (valueType === 'string') {
+          return (
+            <DisplaySchemaPropertyString
+              label={props.schemaObject.properties[key].title}
+              value={value}
+              key={`property_string_${key}`}
+            />
+          )
+        }
+
+        if (valueType === 'boolean') {
+          return (
+            <DisplaySchemaPropertyBoolean
+              label={props.schemaObject.properties[key].title}
+              value={value}
+              key={`property_boolean_${key}_${value}`}
```

Comment:
```suggestion
              key={key}
```
It seems to be putting `value` in the `key` is wrong: unless I'm mistaken, if the value changes, you still want React to match this new value to this DisplaySchemaPropertyBoolean element .

Reply:
that's true
but I don't think it's an issue here because since we switch from the form display and the "static" display to change the value, that particular case won't happen

will remove it anyway, it's more of an artifact of my key management research